baremetal/src/VMem.cc

```diff
@@ -87,6 +87,27 @@ void ebbrt::vmem::MapMemory(Pfn vfn, Pfn pfn, uint64_t length) {
                     });
 }
 
+// traverses per core page table and backs vaddr with physical pages
+// in pfn
+void ebbrt::vmem::MapMemoryLarge(uintptr_t vaddr, Pfn pfn, uint64_t length) {
+  auto pte_root = Pte(ReadCr3());
+  TraversePageTable(
+      pte_root, vaddr, vaddr + length, 0, 4,
+      [=](Pte& entry, uint64_t base_virt, size_t level) {
+        kassert(!entry.Present());
+        entry.Set(pfn.ToAddr() + (base_virt - vaddr), level > 0);
+        std::atomic_thread_fence(std::memory_order_release);
+      },
+      [](Pte& entry) {
+        auto page = page_allocator->Alloc();
+        kbugon(page == Pfn::None());
+        auto page_addr = page.ToAddr();
+        new (reinterpret_cast<void*>(page_addr)) Pte[512];
+        entry.SetNormal(page_addr);
+        return true;
+      });
+}
+
```

Comment:
Yeah the only difference is that it takes a uintptr_t vaddr compared to before which was using a ebbrt::Pfn, this was because I stored this in a std::map and if I just did ebbrt::Pfn::LDown(faulted_address), it didn't give me the right address since it need to to a .ToLAddr() and the original MapMemory used ToAddr() which was assuming a 4k page size